src/utils/datetime/validator.test.ts

```diff
@@ -0,0 +1,95 @@
+    ).toBeTruthy()
+    expect(isValid('1999/02/09 23:00:00', 'YYYY/MM/DD HH:mm:ss')).toBeTruthy()
+    expect(
+      isValid('1999/02/09 11:00:00 PM', 'YYYY/MM/DD hh:mm:ss a')
+    ).toBeTruthy()
+    expect(isValid('23:00', 'HH:mm')).toBeTruthy()
+    expect(isValid('11:00 PM', 'hh:mm a')).toBeTruthy()
+    expect(isValid('23:00:00', 'HH:mm:ss')).toBeTruthy()
+    expect(isValid('11:00:00 PM', 'hh:mm:ss a')).toBeTruthy()
+    expect(isValid('23:00:00 -0700', 'HH:mm:ss ZZ')).toBeTruthy()
+    expect(isValid('11:00:00 PM -0700', 'hh:mm:ss a ZZ')).toBeTruthy()
+    expect(isValid('23:00:00.000', 'HH:mm:ss.sss')).toBeTruthy()
+    expect(isValid('11:00:00.000 PM', 'hh:mm:ss.sss a')).toBeTruthy()
+    expect(
+      isValid('February 9, 1999 23:00:00', 'MMMM D, YYYY HH:mm:ss')
+    ).toBeTruthy()
+    expect(
+      isValid('February 9, 1999 11:00:00 PM', 'MMMM D, YYYY hh:mm:ss a')
+    ).toBeTruthy()
+    expect(
+      isValid(
+        'Tuesday, February 9, 1999 23:00:00',
+        'dddd, MMMM D, YYYY HH:mm:ss'
+      )
+    ).toBeTruthy()
+    expect(
+      isValid(
+        'Tuesday, February 9, 1999 11:00:00 PM',
+        'dddd, MMMM D, YYYY hh:mm:ss a'
+      )
+    ).toBeTruthy()
+  })
+  it('should return false on invalid date formats', function() {
+    expect(isValid('1999-02-09 23:00', 'YYYY-MM-DD HH:mm:ss')).toBeFalsy()
+    expect(
+      isValid('1999-02-09 12:00:00 PM', 'YYYY-MM-DD hh:mm:ss a ZZ')
+    ).toBeFalsy()
+    expect(
+      isValid('09/02/1999 23:00:00', 'DD/MM/YYYY HH:mm:ss.sss')
+    ).toBeFalsy()
+    expect(
+      isValid('09/02/1999 11:00:00.000', 'DD/MM/YYYY hh:mm:ss.sss a')
+    ).toBeFalsy()
+    expect(
+      isValid('02/09/1999 23:00:00', 'MM/DD/YYYY HH:mm:ss.sss')
+    ).toBeFalsy()
+    expect(
+      isValid('02/09/1999 11:00:00.000', 'MM/DD/YYYY hh:mm:ss.sss a')
+    ).toBeFalsy()
+    expect(isValid('1999/02/09 23:00', 'YYYY/MM/DD HH:mm:ss')).toBeFalsy()
+    expect(isValid('1999/02/09 11', 'YYYY/MM/DD hh:mm:ss a')).toBeFalsy()
+    expect(isValid('23', 'HH:mm')).toBeFalsy()
+    expect(isValid('11:00', 'hh:mm a')).toBeFalsy()
+    expect(isValid('23:00', 'HH:mm:ss')).toBeFalsy()
+    expect(isValid('11:00', 'hh:mm:ss a')).toBeFalsy()
+    expect(isValid('23:00:', 'HH:mm:ss ZZ')).toBeFalsy()
+    expect(isValid('11:00:00 ', 'hh:mm:ss a ZZ')).toBeFalsy()
+    expect(isValid('23:00:00', 'HH:mm:ss.sss')).toBeFalsy()
+    expect(isValid('11:00:00.0', 'hh:mm:ss.sss a')).toBeFalsy()
+    expect(
+      isValid('February 9, 1999 23:00', 'MMMM D, YYYY HH:mm:ss')
+    ).toBeFalsy()
+    expect(
+      isValid('February 9, 1999 11:00 PM', 'MMMM D, YYYY hh:mm:ss a')
+    ).toBeFalsy()
+    expect(
+      isValid('Tuesday, February 9, 1999 23:00', 'dddd, MMMM D, YYYY HH:mm:ss')
+    ).toBeFalsy()
+    expect(
+      isValid(
+        'Tuesday, February 9, 1999 11:00:00',
+        'dddd, MMMM D, YYYY hh:mm:ss a'
+      )
+    ).toBeFalsy()
+  })
```

Comment:
nice tests! i don't think you need to change anything here, but just for your own knowledge, there's a school of thought that says we should only have [one assertion per test](https://softwareengineering.stackexchange.com/questions/7823/is-it-ok-to-have-multiple-asserts-in-a-single-unit-test) for a number of reasons - the main being that multiple assertions often confuse the intent of code. another is that when a test fails, if there's one assertion per test, then which test the failure occurred on is obvious and it tells us about the nature of the failure. i don't think these are reasons to change your tests when they pass, but just for your own understanding of testing methodologies, i think you should familiarize yourself with them

Reply:
Got it! I'll read that! That also makes sense, if it were to fail, it'd be hard to know what case failed exactly.